Bottom: e9470eb809e5adf55990fac1b4e16273af3dedb5
Top:    947c8f3fcc13efe3e71abf0b6444f5b8ac71fef4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:31:27 +0000

Debug stuff


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index f711bfe..ab55a5c 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1238,19 +1238,21 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
         {
           fprintf (stderr, "Accumulating elements from layer %i\n", GetLayerNumber (PCB->Data, layer));
 
-//          r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
-//          r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
-//          r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
+          r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
+          r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
+          r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
           r_search (layer->polygon_tree, &bounds, NULL, polygon_copper_callback, &info);
         }
       END_LOOP;
+#endif
 
-//      fprintf (stderr, "Accumulating pin + via pads\n");
-//      r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
-//      r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
+#if 1
+      fprintf (stderr, "Accumulating pin + via pads\n");
+      r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
+      r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
 #endif
 
-#if 0
+#if 1
       if (group == top_group ||
           group == bottom_group)
         {
diff --git a/src/hid/step/step.c b/src/hid/step/step.c
index 47499ce..52182c0 100644
--- a/src/hid/step/step.c
+++ b/src/hid/step/step.c
@@ -152,7 +152,11 @@ step_do_export (HID_Attr_Val * options)
   if (filename == NULL)
     filename = "pcb-out.step";
 
+#if 1
   board_outline_list = object3d_from_board_outline ();
+#else
+  board_outline_list = NULL;
+#endif
 
   board_outline = board_outline_poly (true);
   piece = board_outline;
diff --git a/src/polygon1.c b/src/polygon1.c
index 039b9b5..b662cb8 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -39,7 +39,7 @@
       are marked
 */
 
-#undef NDEBUG
+//#undef NDEBUG
 #include	<assert.h>
 #include	<stdlib.h>
 #include	<stdio.h>
@@ -110,13 +110,13 @@ int vect_inters2 (Vector A, Vector B, Vector C, Vector D, Vector S1,
 
 #define error(code)  longjmp(*(e), code)
 
-#define DEBUG_INTERSECT
-#define DEBUG_LABEL
-#define DEBUG_ALL_LABELS
-#define DEBUG_JUMP
-#define DEBUG_GATHER
-#define DEBUG_ANGLE
-#define DEBUG
+#undef DEBUG_INTERSECT
+#undef DEBUG_LABEL
+#undef DEBUG_ALL_LABELS
+#undef DEBUG_JUMP
+#undef DEBUG_GATHER
+#undef DEBUG_ANGLE
+#undef DEBUG
 #ifdef DEBUG
 #define DEBUGP(...) pcb_fprintf(stderr, ## __VA_ARGS__)
 #else
@@ -2964,9 +2964,13 @@ poly_Boolean_free (POLYAREA * ai, POLYAREA * bi, POLYAREA ** res, int action)
 
   *res = NULL;
 
+#if 0
   /* Make copies for tracking polygon parentage (DEBUG) */
   if (!poly_M_Copy0 (&a_copy, a) || !poly_M_Copy0 (&b_copy, b))
       return err_no_memory;
+#endif
+  a_copy = NULL;
+  b_copy = NULL;
 
   /* Move the parentage information over onto the copy */
   if (a_copy != NULL)
